master/buildbot/data/base.py

```diff
@@ -88,6 +88,9 @@ def control(self, action, args, kwargs):
     def startConsuming(self, callback, options, kwargs):
         raise NotImplementedError
 
+    def __repr__(self):
+        return "endpoint for" + self.pathPatterns
```

Comment:
don't we try to avoid string summation?  and use interpolation instead?

[nit] missing space